Domains: Google Apps: Show a domain warning for accepting Gapps' ToS

klimeryk opened the pull request:
The purpose of this PR is to add a notice in Domain Management notifying the user that has purchased Google Apps account(s) that he/she has to log in to them in order to "activate" them (bascially, accept Google's Terms of Service).

One domain, one account needs verifying:
<img width="744" alt="screen shot 2016-05-28 at 05 51 24" src="https://cloud.githubusercontent.com/assets/3392497/15625044/b5772382-2499-11e6-8a0b-d7ea4de9b928.png">

One domain, multiple accounts (notice `emails` and their list - much plural, such wow):
<img width="746" alt="screen shot 2016-05-28 at 06 04 06" src="https://cloud.githubusercontent.com/assets/3392497/15625072/3a541a1a-249a-11e6-88e8-1d24b78fc586.png">


Multiple domains, multiple accounts need verifying:
<img width="750" alt="screen shot 2016-05-28 at 06 10 01" src="https://cloud.githubusercontent.com/assets/3392497/15625080/dcf94452-249a-11e6-91b9-9d22a8081dc4.png">


The `Learn more` link leads to https://en.support.wordpress.com/adding-google-apps-to-your-site/#completing-sign-up.
The `Log in` links should lead to a prefilled Google's login form - the form is prefilled with the first unverified email from the domain. Why? Because logging in into one of the unverified accounts, verifies the rest of them for the given domain.

These notices should show up in Domain Management:
* the overview of domains for the given site
* the selected domain
* overall Google Apps view
* `Email` view of the selected domain

Additionally, the severity of the notice is based on the date the subscription started:
* younger than 7 days - `info`
* between 7 and 21 days - `warning`
* after 21 days - `error`

Note that the highest severity should trump all below, so if you have one domain that is `info` and one that is `warning`, the overall notice for these domains should be `warning` (while on the individual domain's views it should be `info` and `warning`, respectively).

Added analytics recording the following info for each click on `Log in`:
* `domain`
* `user` - user's email
* `severity` - the severity of the notice, useful to see what's the most effective severity
* `isMultipleDomains` - was the notice for multiple domains or for one selected domain
* `section` - the section where the notice was displayed, either `domain-management` or `google-apps`

Let me know if that's enough info, if I've missed something and if I've done this analytics thing correctly at all :grin:

Of course, if you don't have Gapps, or all your accounts have accepted the ToS, none of these notices should be shown.

Backend changes are backwards compatible and have been merged already, so this can be tested without a sandbox 👍 

/cc @aidvu @umurkontaci @matthusby @rads for code review
/cc @ranh for converting my derp-speak to something understandable by users and concise :wink:
/cc @Lochlaer @BandonRandon because you're awesome like that ❤️

BandonRandon commented:
To be honest, I'm not sure how the suspension process works. It's my understanding that an account is "suspended" from purchase until they first log in no matter the time that takes.Hopefully @Lochlaer can clarify things a bit for everyone. 

I have gone ahead and tested the patch with a new Google Apps purchase. It works as described above but here's some feedback. 

@rahn may also have some comments about the copy. I just wanted to mention I don't love "fix now." makes it sound like something's broken what about:
>Please log in and accept Google's Terms of Service to prevent Google Apps suspension. Log in Now
![screenshot_19-05-2016-22 43 45](https://cloud.githubusercontent.com/assets/150348/15418329/3d388302-1e13-11e6-8594-496028dbba33.png)

I'm also wondering if there is a technical reason why we are not just sending the user to Google's log in screen from the first two alert locations (Domain Management - All and Domain Management - Single) Seems like this could save a click. The advantage of having both alerts is that hopefully the user will read them both but one longer alert may be better here. 

I also think it's a good idea to let the user know where to find the log in information. I do know this is included in the "Learn More" link but wonder if we should also add it to the message:

> Log in to your hello@example.com Google Apps account to accept Google's Terms of Service to avoid suspension. Check your WordPress.com email, example@gmail.com for log in details. Learn More

Speaking of the Learn More link in the alert, can we update the location to: https://en.support.wordpress.com/adding-google-apps-to-your-site/#completing-sign-up

This goes over the completing sign up steps we're trying to get the user to complete with these steps. 

>Bonus question/idea: should I add some tracking events to see how successful these notices are?

I think adding a track to see how many people click at each step could be helpful and potentially help us remove/streamline this flow in the future. (For example, do people log in more from the domains manager or email manager, which A/B test of text works better etc)

Lochlaer commented:
Can we change the copy here:

<img width="744" alt="b2d7b878-1e3e-11e6-9803-f753aa59c735" src="https://cloud.githubusercontent.com/assets/6569475/15431936/b31965d8-1e71-11e6-89c5-9b3afce93990.png">

To something like this:

> Urgent! Log in and activate your Google Apps subscription for the domain example.com now. 

@BandonRandon is correct here:

> an account is "suspended" from purchase until they first log in no matter the time that takes.

Re: this:

> `Fix now` should redirect to the domain-specific Google Apps view)

Yes. More specifically, if possible it should redirect to the target URL so that there's no confusion if the user has multiple Google Apps/Gmail accounts. For example:

https://accounts.google.com/AccountChooser?Email=**user@companyname.com**&service=CPanel&continue=https%3A%2F%2Fadmin.google.com%2F**companyname.com**%2FAcceptTermsOfService%3Fcontinue%3Dhttps%3A%2F%2Fmail.google.com%2Fmail%2Fu%2F1

This is the pointer URL that Dennis at Google recommends to avoid the conflicting login pages altogether. ( /cc @matthusby because we talked about this before as a possibility.)

> Bonus question/idea: should I add some tracking events to see how successful these notices are?

Yes, please.

> (Learn more links to https://support.wordpress.com/add-email/adding-google-apps-to-your-site - any suggestions for better target?)

Yes - a new page that's more targeted at users who've already purchased Google Apps. :-) For now, can you please just use the one that @BandonRandon recommends above and then I'll work on a new doc that will be more appropriate? 

Thanks!

ranh commented:
I was not able to get these notices to show up. I could use some more information about this whole process. 

What is the expected flow for getting a new GApps account? Where does accepting the ToS normally fit in? 

Is there anything else we or Google are doing to get users to accept the ToS? Emails? Other notices? 

What does "your account will get suspended soon" mean? How soon? What is the implication of having the account suspended?

From what I understand, accepting the ToS is the first thing one does with GApps, so if a user hasn't accepted the ToS, it means they haven't used their GApps account at all. Is that correct? If so, the ToS is just a technicality, we should focus on getting the account working and getting the user using it

klimeryk commented:
> What is the expected flow for getting a new GApps account? Where does accepting the ToS normally fit in?

When registering a custom domain (through NUX or Domains Management) after selecting the domain from the domain search results you are shown this nudge:

![2016-05-24-171612_1157x562_scrot](https://cloud.githubusercontent.com/assets/3392497/15511293/23328e78-21db-11e6-8abd-e3b969290678.png)

Click `Add Email` to add one or more Google Apps accounts to the domain you are buying (one license = one `whatever@example.com` account).

> Is there anything else we or Google are doing to get users to accept the ToS? Emails? Other notices?

After setting up the account (should not be more than a few minutes), the user gets an email from us titled `[WordPress.com] Google Apps is now set up and activated for your domain "example.com".`:
![2016-05-24-172036_1452x1047_scrot](https://cloud.githubusercontent.com/assets/3392497/15511447/c76792fe-21db-11e6-988d-1c6572742500.png)

At around the same time you get an email from Google titled `You’ve signed up for Google Apps: What’s next?`:
![2016-05-24-172236_934x1151_scrot](https://cloud.githubusercontent.com/assets/3392497/15511494/0c415306-21dc-11e6-9812-ebfc733107da.png)

That is - if everything goes smoothly :)
The common issue is the domain not being verified properly (we automatically add the necessary TXT records and retry the verification a few times, but it still seems to fail from time to time). The user can verify it himself (Google sends an email if you don't verify your domain soon after purchase), but we also have HEs do it.

> What does "your account will get suspended soon" mean? How soon? What is the implication of having the account suspended?

> From what I understand, accepting the ToS is the first thing one does with GApps, so if a user hasn't accepted the ToS, it means they haven't used their GApps account at all. Is that correct? If so, the ToS is just a technicality, we should focus on getting the account working and getting the user using it

I'll let @Lochlaer answer these ones, as she is more familiar with the process, the numbers behind it and how it looks compared to other resellers.

Lochlaer commented:
@ranh 

> What does "your account will get suspended soon" mean? How soon? What is the implication of having the account suspended?

That's why I want to change the copy to something like this:

> Urgent! Log in and activate your Google Apps subscription for the domain example.com now.

It's *already* suspended, so "your account will be suspended soon" doesn't really make sense. The account is automatically in a suspended state upon creation/provisioning until the user logs in for the first time and accepts Google's TOS and reseller agreement. 

Until the user does that, they're unable to use the account and it continues to show as `Suspended - Terms of Service` in the reseller dashboard. 

> From what I understand, accepting the ToS is the first thing one does with GApps, so if a user hasn't accepted the ToS, it means they haven't used their GApps account at all. Is that correct?

That is correct. 

> If so, the ToS is just a technicality, we should focus on getting the account working and getting the user using it

Totally a technicality and probably one of the biggest blockers to product adoption (which Google defines as when the user logs in at least once every 30 days). The idea is to reduce/eliminate the friction from purchase to initial sign-in. 

Hope that clarifies things!

umurkontaci commented:
Can we remove "Urgent !" from the beginning of the message? Just because they didn't accept Google Apps ToS doesn't mean we should scare the hell out of them and make them uncomfortable and indicate something is terribly wrong. I think a positive reinforcement in this case will go a long way.

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/index.jsx:
```diff
@@ -215,6 +215,25 @@ export default React.createClass( {
+				status="is-error"
+				showDismiss={ false }
+				key="pending-gapps-tos-acceptance-domains"
+				text={ this.translate( 'Urgent! You need to accept Google\'s Terms of Service or else your Google Apps accounts will get suspended soon.' ) }>
```
We know the domains, why not show them in the message?

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/index.jsx:
```diff
@@ -215,6 +215,25 @@ export default React.createClass( {
+				key="pending-gapps-tos-acceptance-domains"
+				text={ this.translate( 'Urgent! You need to accept Google\'s Terms of Service or else your Google Apps accounts will get suspended soon.' ) }>
+				<NoticeAction href={ paths.domainManagementEmail( this.props.selectedSite.domain, this.getDomains().length === 1 && domains[0].name ) }>
+					{ this.translate( 'Fix now' ) }
```
This breaks Calypso for me. `googleAppsUsers` has a null item in the state tree and breaks everything, including navigating away. 

![image](https://cloud.githubusercontent.com/assets/991022/15522426/1b3e4efa-21c7-11e6-9dec-74f8b374196a.png)

Lochlaer commented:
> Can we remove "Urgent !" from the beginning of the message? Just because they didn't accept Google Apps ToS doesn't mean we should scare the hell out of them and make them uncomfortable and indicate something is terribly wrong. I think a positive reinforcement in this case will go a long way.

I'd really like to have *something* that conveys the, well, *urgency* of logging in and thus activating their account. The user has paid a lot of money for this particular product, so we want to make sure that this last step is top-of-mind when they log into their dashboard so that they can start using it right away. 

Alternative suggestions welcome! Maybe `Important:`?

umurkontaci commented:
This is a nice to have:
![image](https://cloud.githubusercontent.com/assets/991022/15522910/c8a4e164-21ca-11e6-8e44-802655a62d68.png)

Instead of just showing the error with a notice above, we can indicate that something is wrong inside the user row as well (maybe something with a red exclamation mark or similar)

umurkontaci commented:
> I'd really like to have something that conveys the, well, urgency of logging in and thus activating their account. The user has paid a lot of money for this particular product, so we want to make sure that this last step is top-of-mind when they log into their dashboard so that they can start using it right away.

Just because you paid good money doesn't mean you should *urgently* set it up though :) I definitely agree we need to max ToS acceptance, but I don't think it's a good idea to overwhelm the user. They buy a domain and Google Apps from us, and they receive 2 urgent warnings (ICANN confirmation and Google Apps Tos) and one yellow warning (new domain setting up) as if all hell broke loose.

I think this is too much. 

Couple of things to consider in here; it's still early to assume the user knows what Google Apps is. I thought Google Apps was Google Maps, Google+ and such before. Almost everyone buys it for email and we can use that. 

How about something along the lines of "Finish setting up your email admin@example.com, you will *not* receive or send messages until you finish setting up."

BandonRandon commented:
I like where @umurkontaci is going with this positive reinforcement thing. What about something like we do for domain mappings/registrations (showing the yellow warning instead of the red error)
>  Almost set! To start using your email admin@example.com please log in and accept Google's ToS. Learn More

You also make a good point about all the warnings and emails we send. I wonder if we can do something like delay this message for 24 hours after signup (or until the yellow domain alert goes away). If the user still hasn't logged in then show the error. 

I'm also making the assumption the user will be setting their own password via the flow. If this is not the case we should let them know where to find the log in information. I would expect this may be where some of the confusion is coming from. Users who _want_ to log in but don't know how. 

If we are still setting a temporary password I would change the message to something more along the lines of:
> Almost set! to start receiving email at admin@example.com please follow the instructions sent to user_email@example.com.

umurkontaci commented:
> I like where @umurkontaci is going with this positive reinforcement thing. What about something like we do for domain mappings/registrations (showing the yellow warning instead of the red error)
> 
> Almost set! To start using your email admin@example.com please log in and accept Google's ToS. Learn More

I like this copy better than mine!




> I wonder if we can do something like delay this message for 24 hours after signup (or until the yellow domain alert goes away). If the user still hasn't logged in then show the error.

Not sure if the signup timestamp information for Google Apps accounts is readily available, but for domains it is. We can make use of that information to progressively increase the "urgency" of the message as well. 

Also, instead of showing these as notices – which was done because it was easy – we can make a small onboarding process that guides the user to set up things; similar to some companies do on post-signup with progress bars and steps. It's an idea for the future though, we can talk about it somewhere else to keep this PR in orbit.

BandonRandon commented:
> make use of that information to progressively increase the "urgency" of the message

I love this idea! No need to sound all alarms from the start, but multiple alerts may allows the user to know that this needs to be done. 

>  – we can make a small onboarding process that guides the user to set up things... we can talk about it somewhere else to keep this PR in orbit.

I agree that this is best for a conversation elsewhere. I just wanted to say ideally we'd have both. Combining both the drip/welcome stuff that other teams are doing with the alerts here. Ultimately our goal is to to make it as simple for users to be aware how WordPress.com works and the steps they need to take. Just like different people have different communication styles, they may respond to various alerts differently.

Lochlaer commented:
I like this copy too:

> Almost set! To start using your email admin@example.com please log in and accept Google's ToS. Learn More

Corrected to:

> Almost set! To start using your email admin@example.com, please log in and accept Google's Terms of Service. Learn More.

Question for @klimeryk : Can we change the `Learn More` text to `Click Here` or `Log In Now` and then link that button to the pointer URL (with the custom domain included) I mentioned in the thread above? 

A couple of other considerations:

* I like the idea of staggering the alerts. We also need to consider that part of the TOS acceptance campaign includes a series of emails (schedule TBD) to send to users reminding them if they haven't logged in yet, so I think a yellow alert at the outset is appropriate, which can be changed to red after a day or so. 
* We also need to consider that part of the roadmap includes changing the purchase flow so that users will be able to create their own passwords/user accounts post-purchase, so this flow we're discussing may or may not change (depending on if/when we roll out that change. 

I had other comments about the TOS acceptance campaign in Slack, but I don't want to delay deployment of this. Let's Track it and see how things go!

@ranh Comments/feedback welcome on the copy!

BandonRandon commented:
> Can we change the Learn More text to Click Here or Log In Now and then link that button to the pointer URL (with the custom domain included) I mentioned in the thread above? 
@Lochlaer, are you suggesting that in the alert we nix the "Learn More" link and change it to Click Here" or is this just for the "Log in" text to redirect the pointer URL above. 

I don't think we should get rid of the link to the support document found under "Learn More" as this could help some users if they have trouble with the log in link.

klimeryk commented:
So much great ideas - thanks guys and gals! I especially like the positive wording of @umurkontaci's suggestion and the progressive increase in urgency of the notice. I'll start working on implementing those changes ASAP (the copy can be tweaked easily if needed).

ranh commented:
I like this direction. I think we can still lose the reference to suspension and the ToS, we're the only ones who care about that. The focus should be on using the product.

How about this?

> You're almost there! To activate your email admin@example.com, please log in to Google Apps and finish setting it up. Learn More.

klimeryk commented:
> How about this?
> 
> > You're almost there! To activate your email admin@example.com, please log in to Google Apps and finish setting it up. Learn More.

Love it 👍 We'll see how it fares and if there's no significant uptake in ToS acceptance we can iterate on this copy.

Just to be thorough - could you also provide a version for two or more unverified emails? I'd love to provide a `Log in` link somewhere for them in the notice too.

ranh commented:
Sorry, forgot about that variation! Thanks for the ping. How about: 

> You're almost there! To activate your new email addresses, please log in to Google Apps and finish setting them up. Learn More.

I have to ask though, is this supposed to be done by the same person? For example, if I add 10 new accounts, I have to log in separately and activate each one?

matthusby commented:
> I have to ask though, is this supposed to be done by the same person?

It only needs to be done once per account.

klimeryk commented:
I think what Ran was asking about is does the initial login (accepting of ToS) has to be done by the same person or not. I think it depends - the subscription's owner gets all the emails from Google (the ones with the temporary password, see above), but he/she can forward them to someone else, effectively giving them the account. But I'd love a confirmation/better explanation of how multiple accounts per domain work :)

matthusby commented:
It needs to be done once per account.  Someone needs to login as the admin and accept the TOS etc.  Most of the time this will be the person that made the purchase, but I guess they could send the emails to someone else.

Lochlaer commented:
> It needs to be done once per account. Someone needs to login as the admin and accept the TOS etc. Most of the time this will be the person that made the purchase, but I guess they could send the emails to someone else.
 
Seconding this response. All licenses that a user purchases under the same account is automatically classified as a *Super Admin* account initially, and that's all that matters when it comes to logging in and accepting the TOS for the entire account. So the user can log into *one* (any) of those accounts and accept the TOS, which then automatically accepts it for *all* the licenses under that account.

Anyone who has the login credentials can log into the account and accept the TOS -- this is not at all dependent on ownership of the upgrade within WPCOM. However, for security reasons it should only be the purchaser who gets the alert/notification. 

Once the sub owner logs in and accepts the TOS, they can go to their Google Apps admin dashboard (admin.google.com) and manage the login creds for all other licenses under that account: send the passwords/logins for the other licenses to other users, etc. 

Hope that makes sense, @ranh!

ranh commented:
Thanks for the clarifications. If I understand correctly, by "Account" you mean essentially the whole domain, and by "license" you mean the individual email addresses? If that is correct, then based on @Lochlaer's comment above, it sounds like there is no need to show a list of multiple email addresses and Log In links to confirm the ToS. Just showing one will do?

Lochlaer commented:
@ranh 

> by "Account" you mean essentially the whole domain, and by "license" you mean the individual email addresses?

Basically, yes. 

> it sounds like there is no need to show a list of multiple email addresses and Log In links to confirm the ToS. Just showing one will do?

For the purposes of accepting the TOS,  yeah, they only need to log into one of those licenses. Might be a good idea for it to be the first one that they buy -- chances are, that's the "real" Super Admin (i.e., the account owner's license). 

It would be good to show all the other licenses once the account is active (not Suspended) so that the Admin can manage it all from that part of their dashboard.

umurkontaci commented:
This has a good example on what to do: https://github.com/Automattic/wp-calypso/blob/master/client/components/search/test/index.jsx#L20

The core ideas is you want to register a mock before importing the component you are about to the test. The example I gave you is more or less the convention we follow along with some helper functions. 

At a bare minimum you need

```
import mockery from 'mockery'
describe('test', () => {
  let Component;
  before( () => {
    mockery.registerMock('lib/analytics', {})
    Component = require('../component');
  } );
  after(() => {
    mockery.deregisterMock('lib/analytics');
  } );

klimeryk commented:
Thanks a lot, @umurkontaci! :bow: All green, ready for another review!

BandonRandon commented:
Worked in testing:
![screenshot_31-05-2016-21 56 43](https://cloud.githubusercontent.com/assets/150348/15698575/b6d43b82-277a-11e6-84dc-6a084714837f.png)

umurkontaci commented on client/lib/mixins/analytics/index.js:
```diff
@@ -525,6 +525,26 @@ const EVENTS = {
+						isMultipleDomains,
+						section
+					}
+				);
```
I think it'd be nice to record impressions as well so we have a proper conversion measure

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/pending-gapps-tos-notice.jsx:
```diff
@@ -0,0 +1,118 @@
+			return domain => this.moment( domain.googleAppsSubscription.subscribedDate ).isBefore( this.moment().subtract( days, 'days' ) );
+		};
+
+		if ( this.props.domains.some( subscribedDaysAgo( 21 ) ) ) {
```
Nice!

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/pending-gapps-tos-notice.jsx:
```diff
@@ -0,0 +1,118 @@
+				showDismiss={ false }
+				key="pending-gapps-tos-acceptance-domain"
+				text={ this.translate(
+					'You\'re almost there! To activate your email {{strong}}%(emails)s{{/strong}}, please log in to Google Apps and finish setting it up. {{learnMoreLink}}Learn More.{{/learnMoreLink}}',
```
This PR lingered for long so I don't want to create a blocker, but I think it'd be great to have a different copy for different severities: 

![image](https://cloud.githubusercontent.com/assets/991022/15762585/7473df36-28d3-11e6-9a76-637fb4ee3423.png)

This kinda feels awkward. It's red but it's suggestive.

umurkontaci commented on client/my-sites/upgrades/domain-management/email/google-apps-users-card.jsx:
```diff
@@ -27,11 +28,14 @@ const GoogleAppsUsers = React.createClass( {
 
-	canAddUsers() {
-		const domainsInContext = this.props.selectedDomainName
+	getDomains() {
```
It's not clear what this function does on the first look. It's named `getDomains`, uses props `selectedDomainName` and `domains` and a `getSelectedDomain` function.

Although I know what it does since I've been playing around with this, adding a bit of comment would be helpful for others.

klimeryk replied:
I was "inspired" by the [`getDomains` function from `domain-warnings`](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/upgrades/components/domain-warnings/index.jsx#L65). But you do have a point - maybe something like `getDomainsAsList`?

umurkontaci replied:
I think just a line comment will do

umurkontaci commented:
I've tested in on Domains list and edit pages and they are working as they should.

Left a couple of comments on code, I think tracking impressions is the most important thing there. I'm OK if you want to follow up with the other ones another day.

klimeryk commented:
@umurkontaci - thanks for the review, very good points - hopefully I've addressed them all :)

I even caught one old bug related to tracking in Google Apps view (see https://github.com/Automattic/wp-calypso/pull/5476/commits/90991718a4df9ddda39e1e1e19ee4d616784bb88) :grin:

As for the different copy for different severities - I've changed the exclamation that the notice starts based on the severity, I hope that's enough. So, `info` => `You're almost done!`, `warning` => `Attention!`, `error` => `Urgent!`.

umurkontaci commented on client/my-sites/upgrades/domain-management/email/google-apps-users-card.jsx:
```diff
@@ -28,14 +28,14 @@ const GoogleAppsUsers = React.createClass( {
 	},
 
-	getDomains() {
+	getDomainsAsList() {
```
🎉

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/pending-gapps-tos-notice.jsx:
```diff
@@ -18,10 +18,18 @@ const PendingGappsTosNotice = React.createClass( {
 	mixins: [ analyticsMixin( 'domainManagement', 'googleApps' ) ],
 
 	propTypes: {
+		site: React.PropTypes.oneOfType( [
```
Is this `selectedSite`?

umurkontaci commented on client/lib/mixins/analytics/index.js:
```diff
@@ -525,6 +525,43 @@ const EVENTS = {
+					{
+						site: siteSlug,
+						domain_name: domainName
+					}
```
I think we should make the properties of the two events as similar as possible. We should be able to add as much detail as in the click event.

umurkontaci commented on client/lib/mixins/analytics/index.js:
```diff
@@ -525,6 +525,43 @@ const EVENTS = {
+				analytics.tracks.recordEvent(
+					'calypso_domain_management_google_apps_pending_account_log_in_click',
+					{
+						domain,
```
Here it is referred as `domain` and the in the other one, it's `domain_name`. I think the convention is to use `domain_name`.

klimeryk replied:
Good point! 👌

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/pending-gapps-tos-notice.jsx:
```diff
@@ -0,0 +1,152 @@
+	mixins: [ analyticsMixin( 'domainManagement', 'googleApps' ) ],
+
+	propTypes: {
+		site: React.PropTypes.oneOfType( [
```
It seems we are not using `site` for anything else than tracking, do you think it'd be a good idea to just pass the slug and call it `siteSlug`?

umurkontaci commented:
General testing shows it looks like it's working, but I wasn't able to test anything under Google Apps tab as it is broken for me.

I think that piece was tested before so that part should be working :)

klimeryk commented:
Thanks for the feedback - I've addressed the issues you've pointed out and it's ready for another review :bow:

umurkontaci commented on client/my-sites/upgrades/components/domain-warnings/index.jsx:
```diff
@@ -221,7 +221,7 @@ export default React.createClass( {
 				domain.googleAppsSubscription.pendingUsers &&
 				domain.googleAppsSubscription.pendingUsers.length !== 0 );
-		return pendingDomains.length !== 0 && <PendingGappsTosNotice key="pending-gapps-tos-notice" site={ this.props.selectedSite } domains={ pendingDomains } section="domain-management" />;
+		return pendingDomains.length !== 0 && <PendingGappsTosNotice key="pending-gapps-tos-notice" siteSlug={ this.props.selectedSite.slug } domains={ pendingDomains } section="domain-management" />;
```
Can `selectedSite` be `false` here?

klimeryk replied:
Yes and no - it can be passed `false` from `CurrentSite` (which, despite its name, is also shown when browsing all your sites), but the rule for showing this notice is not whitelisted in that context. Nevertheless, to be on the safe side, I've added the check - good catch!

umurkontaci commented:
Changes look good, did a partial test. 

I have one comment on existence of `selectedSite`, add that, do a rebase with master and it's good to go. No need for another review 
👍
